pkg/config/conversion_test.go

```diff
@@ -0,0 +1,53 @@
+package config_test
+
+import (
+	"testing"
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"istio.io/istio/pkg/config"
+)
+
+// Define a sample Spec struct and its proto representation
+type TestSpec struct {
+	Key string
+}
+
+func (t *TestSpec) Reset()        { *t = TestSpec{} }
+func (t *TestSpec) ProtoMessage() {}
+
+func TestPilotConfigToResource(t *testing.T) {
+	spec := &TestSpec{Key: "test-value"}
+	creationTime := time.Now()
+
+	testConfig := &config.Config{
+		Meta: config.Meta{
+			Name:              "test-name",
+			Namespace:         "test-namespace",
+			CreationTimestamp: creationTime,
+			ResourceVersion:   "1",
+			Labels: map[string]string{
+				"app": "test-app",
+			},
+			Annotations: map[string]string{
+				"annotation-key": "annotation-value",
+			},
+		},
+		Spec: spec,
+	}
+
+	resource, err := config.PilotConfigToResource(testConfig)
+	require.NoError(t, err, "Expected no error from PilotConfigToResource")
+
+	assert.NotNil(t, resource, "Expected non-nil resource")
```

Comment:
Just use `assert.Equals(&mcp.Resource { ... })` using the isito assert library instead of testify. 

Asserting each individual field is way harder to read and actually not as correct, since it wouldn't catch us adding extra unexpected fields